preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/bdtopo/BDTopoGISLayers.groovy

```diff
@@ -112,7 +112,37 @@ IProcess importPreprocess(){
             def tmp_imperv_surface_activite = 'TMP_IMPERV_SURFACE_ACTIVITE_' + uuid
             def input_impervious = 'INPUT_IMPERVIOUS'
 
-            // If the impervious tables does not exists, we create corresponding empty tables
+
+            // -------------------------------------------------------------------------------
+            // Check if the input files are present
+
+            // If the IRIS_GE table does not exist or is empty, then the process is stopped
+            if(!datasource.hasTable(tableIrisName) || datasource.getTable(tableIrisName).isEmpty()){
+                logger.error 'The process has been stopped since the table IRIS_GE does not exist or is empty'
+                return null}
+
+            // If the following tables does not exists, we create corresponding empty tables
+            if(!datasource.hasTable(tableBuildIndifName)){
+                datasource.execute("CREATE TABLE $tableBuildIndifName (THE_GEOM geometry(geometry, 2154), ID varchar, HAUTEUR integer);")
+            }
+            if(!datasource.hasTable(tableBuildIndusName)){
+                datasource.execute("CREATE TABLE $tableBuildIndusName (THE_GEOM geometry(geometry, 2154), ID varchar, HAUTEUR integer, NATURE varchar);")
```

Comment:
typed geometry ? polygon ?

Reply:
Ok so I will replace geometry by polygon (and linestring for roads and railways)